projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -395,12 +395,22 @@ public function render_block_item( $related_post, $block_attributes ) {
 			$item_markup .= $date_tag;
 		}
 
-		if ( ( $block_attributes['show_context'] ) && ! empty( $related_post['context'] ) ) {
-			$context_tag = sprintf(
-				'<li class="jp-related-posts-i2__post-context">%1$s</li>',
-				esc_html( $related_post['context'] )
-			);
-
+		if ( ( $block_attributes['show_context'] ) && ! empty( $related_post['bcon'] ) ) {
+			// New render logic, ignoring original context.
```

Comment:
> Could we explain why we're ignoring the original context in this code comment?

This is explained a little in the [related issue](https://github.com/Automattic/jetpack/issues/30673) but the summary is that I think we need a second context for use within the block environment to avoid breaking the existing behaviour.

The current behaviour is to pull in the existing context and render it as needed. You'll see that this kinda works as long as the context isn't "filtered" through any PHP snippets but it breaks things if any filtering is applied. See screenshots in linked issue for that.

> Also, could we add some background information on the source of this context? Is it provided by the related posts API endpoint?

The context is built locally in the PHP function `get_related_post_data_for_post()`. That function sets up data needed for display after getting a list of post IDs from the API in `get_related_post_ids()`.

Based on my understanding of the code paths, we have two options:

1. Keep the original context for legacy rendering and a second context for the block environment (the solution I've coded for here) or…
2. Insert something in the code path so we know if it's a legacy widget or a block and alter the context based on that. This second option has a higher likely hood of breakage as it's used in different scenarios like in the Customizer.

Reply:
Got it; thank you for the context. This approach makes sense to me, though we'll want to update our documentation to reflect this new behavior following a release.

Also, it'd be great to get some of this context into the code as code comments 🙂. I wouldn't mind if we did this in a follow-up PR, though!

Reply:
Added some comments around the 'context' and 'block_context' to help clarify things. Also added a ReadMe file with some notes.